aten/src/ATen/cuda/CUDASparseDescriptors.h

```diff
@@ -0,0 +1,202 @@
+#pragma once
+
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/cuda/CUDADataType.h>
+#include <ATen/cuda/CUDASparse.h>
+#include <ATen/native/LinearAlgebraUtils.h>
+#include <ATen/native/cuda/MiscUtils.h>
+
+#include <c10/core/ScalarType.h>
+
+#if AT_USE_CUSPARSE_GENERIC_API()
+
+namespace at {
+namespace cuda {
+namespace sparse {
+
+namespace {
+
+// If a specific GPU model does not provide native support for a given data
+// type, cuSparse routines return CUSPARSE_STATUS_ARCH_MISMATCH error
+void check_supported_cuda_type(cudaDataType cuda_type) {
```

Comment:
We might want to make this something more specific to sparse, this sounds like something that could easily conflict with another function in another header.

Reply:
This is in an unnamed namespace, so local to this file only.

Reply:
Ah I see! I thought it's generally not a good idea to have anonymous namespaces in headers.

Reply:
Indeed, I'll move it to the `.cpp` file.